packages/mux-uploader/src/mux-uploader-progress.ts

```diff
@@ -87,43 +102,72 @@ class MuxUploaderProgressElement extends globalThis.HTMLElement {
 
     this.svgCircle = this.shadowRoot?.querySelector('circle');
     this.progressBar = this.shadowRoot?.getElementById('progress-bar');
+    this.uploadPercentage = this.shadowRoot?.getElementById('percentage-type');
     this.progressBar?.setAttribute('aria-description', ariaDescription);
   }
 
   connectedCallback() {
     this.setDefaultType();
-    const parent = document.querySelector('mux-uploader');
 
-    parent?.addEventListener('progress', (e) => {
-      // @ts-ignore
-      const percent = e.detail;
-      this.progressBar?.setAttribute('aria-valuenow', `${Math.floor(percent)}`);
+    this.#uploaderEl = getMuxUploaderEl(this);
 
-      switch (this.getAttribute('type')) {
-        case ProgressTypes.BAR: {
-          if (this.progressBar) this.progressBar.style.width = `${percent}%`;
-          break;
-        }
-        case ProgressTypes.RADIAL: {
-          if (this.svgCircle) {
-            // The closer the upload percentage gets to 100%, the closer offset gets to 0.
-            // The closer offset gets to 0, the more we can see the circumference of our circle. (TD).
-            const offset = this.getCircumference() - (percent / 100) * this.getCircumference();
-
-            this.svgCircle.style.strokeDashoffset = offset.toString();
-          }
-        }
-      }
-    });
+    if (this.#uploaderEl) {
+      this.#uploaderEl.addEventListener('uploadstart', this.onUploadStart.bind(this));
+      this.#uploaderEl.addEventListener('reset', this.onReset.bind(this));
+      this.#uploaderEl.addEventListener('progress', this.onProgress.bind(this));
+    }
+  }
 
-    parent?.addEventListener('uploadstart', this.onUploadStart.bind(this));
+  disconnectedCallback() {
+    if (this.#uploaderEl) {
+      this.#uploaderEl.removeEventListener('uploadstart', this.onUploadStart.bind(this));
+      this.#uploaderEl.removeEventListener('reset', this.onReset.bind(this));
+      this.#uploaderEl.removeEventListener('progress', this.onProgress.bind(this));
+    }
   }
 
   onUploadStart() {
     this.progressBar?.focus();
     this.setAttribute('upload-in-progress', '');
   }
 
+  onProgress(e: Event) {
```

Comment:
suggestion(non-blocking): `e: CustomEvent<number>`. Should also remove unnecessary `// @ts-ignore`

Reply:
Will address this in a followup ts cleanup